### PR TITLE
Remove plugins that aren't supposed to be in unsecure build

### PR DESCRIPTION
**Pull request by vjpai**

### CODE REVIEW

**Comment by grpc-testing** [comment]
```
No significant performance differences
```

File: src/core/plugin_registry/grpc_unsecure_plugin_registry.c [hunk 1]
```diff
@@ -71,16 +71,12 @@ void grpc_register_built_in_plugins(void) {
                        grpc_deadline_filter_shutdown);
   grpc_register_plugin(grpc_client_channel_init,
                        grpc_client_channel_shutdown);
-  grpc_register_plugin(grpc_resolver_dns_ares_init,
```
Comment (jtattermusch) [inline src/core/plugin_registry/grpc_unsecure_plugin_registry.c]: Please improve PR description to explain why this should be in unsecure.

File: src/core/plugin_registry/grpc_unsecure_plugin_registry.c [hunk 2]
```diff
@@ -71,16 +71,12 @@ void grpc_register_built_in_plugins(void) {
...
-                       grpc_resolver_dns_ares_shutdown);
   grpc_register_plugin(grpc_resolver_dns_native_init,
                        grpc_resolver_dns_native_shutdown);
   grpc_register_plugin(grpc_resolver_sockaddr_init,
                        grpc_resolver_sockaddr_shutdown);
   grpc_register_plugin(grpc_load_reporting_plugin_init,
                        grpc_load_reporting_plugin_shutdown);
-  grpc_register_plugin(grpc_lb_policy_grpclb_init,
```
Comment (jtattermusch) [inline src/core/plugin_registry/grpc_unsecure_plugin_registry.c]: Please improve PR description to explain why this shouldn't be in unsecure.

**Comment by vjpai** [comment]
The issue was that TF still wasn't building with gRPC at master and these two linkage errors came up. I think we can get the grpclb back in. Maybe we can get cares back in also. I'll try making some changes to BUILD instead to see if that's possible without causing security to be linked in again.

**Comment by ctiller** [comment]
This file is code generated. This change will break at some point.

**Comment by vjpai** [comment]
Yeah, I realized that at some point but forgot to close this PR. Closing now.

**Pull request closed without merging**